sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -620,6 +620,12 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     val df = streaming.join(static, Seq("b"))
     assert(df.isStreaming, "streaming Dataset returned false for 'isStreaming'.")
   }
+
+  test("SPARK-14554: Dataset.map may generate wrong java code for wide table") {
+    val wideDF = sqlContext.range(10).select(Seq.tabulate(1000) {i => ('id + i).as(s"c$i")} : _*)
+    // Make sure the generated code for this plan can compile and execute.
+    wideDF.map(_.getLong(0)).collect()
```

Comment:
Do you know why this test case is super slow? It took more than 5 minutes to finish it. Is this expected?  

```
- SPARK-14554: Dataset.map may generate wrong java code for wide table (5 minutes, 20 seconds)
```

See the link: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/59079/consoleFull